src/components/UserPhotoUploader/UserPhotoUploader.tsx

```diff
@@ -0,0 +1,109 @@
+'use client'
+import React, { useState } from 'react'
+import styles from './UserPhotoUploader.module.scss'
+import { Button } from '../Button/Button'
+import { PutBlobResult } from '@vercel/blob'
+import { useSession } from 'next-auth/react'
+import { useFormikContext } from 'formik'
+import { CreateProfileFormValues } from '../CreateProfileForm/CreateProfileFormWrapper'
+
+export const UserPhotoUploader = () => {
+  const { setFieldValue, values } = useFormikContext<CreateProfileFormValues>()
+  const [selectedFile, setSelectedFile] = useState<File | null>(null)
+  const [userImage, setUserImage] = useState<string>(values.avatarUrl)
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0]
+    if (file) {
+      setSelectedFile(file)
+      const reader = new FileReader()
+      reader.onloadend = () => {
+        setUserImage(reader.result as string)
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const { data: session } = useSession()
+  const importFromGithub = async () => {
+    setUserImage(session?.user?.image || '')
```

Comment:
**nitpick:** We should check `if (!sessionImage) throw Error("No Github image found")`